### ISSUE TITLE
Yama k550

### ISSUE THREAD

**Issue by RusticCraftsman** [GrbavaCigla/opendragon#6]
So... i have a yama k550, its a keyboard and i would like to add support of it, i also would like to contribute to the code which currently i dont understand

**Comment by GrbavaCigla** [comment]
Hi, driver currently doesn't support keyboards and I planning to rewrite large parts of the code, but I think I can add support for it. Do you know how to capture USB packets with wireshark?

Edit: Also if you want to try and contribute yourself, read this: https://github.com/GrbavaCigla/opendragon#sources

**Comment by RusticCraftsman** [comment]
no, i dont know how to, sorry for the late response, do you have any sources so i could learn how to?

**Comment by RusticCraftsman** [comment]
> Edit: Also if you want to try and contribute yourself, read this: https://github.com/GrbavaCigla/opendragon#sources
did not understand this, how am i supposed to contribute based on the sources?

**Comment by GrbavaCigla** [comment]
> no, i dont know how to, sorry for the late response, do you have any sources so i could learn how to?

Do you have a windows machine by any chance? It is very simple, install official redragon drivers on windows machine and install Wireshark (simple installer, just be sure to tick on USBPcap support). After that, select usb port and start capturing what happens when you change something in the driver.

> did not understand this, how am i supposed to contribute based on the sources?

Well currently you don't :D. Driver is tied to mice, so I would have to inspect the data to understand how keyboards work.
